R/setkey.R

```diff
@@ -353,6 +354,9 @@ CJ <- function(..., sorted = TRUE, unique = FALSE)
                 l[[i]] = rep.int(rep.int(y, times = rep.int(x[i], n[i])), times = nrow/(x[i]*n[i]))
             if (any(class(l[[i]]) != j[[i]]))
                 setattr(l[[i]], 'class', j[[i]]) # reset "Date" class - rep.int coerces to integer
+            if (any(!sapply(tzones, is.null))) {
```

Comment:
Are the tzone for all columns being looked at here,  within a loop through columns?   It doesn't seem quite right to me but I haven't debugged or run it.   See next comment on tests.